src/apollo/utils.ts

```diff
@@ -0,0 +1,31 @@
+import { authenticateWalletWithRetry } from '~auth';
+
+export async function mutateWithAuthRetry(
+  mutationFn: () => Promise<any>,
+  maxRetries = 3,
+) {
+  let attempts = 0;
+  let result: any;
+  while (attempts < maxRetries) {
```

Comment:
Hey @iamsamgibbs just a quick thought – since we already have `exponential-backoff` lib in our package.json, would it make sense to refactor this while loop with that library?

Reply:
Oh interesting, I didn't realise we had this package. I will take a look.